packages/builder/src/pages/builder/portal/manage/groups/_components/AppAddModal.svelte

```diff
@@ -0,0 +1,51 @@
+<script>
+  import { Body, ModalContent, Select } from "@budibase/bbui"
+  import { apps, groups } from "stores/portal"
+  import { roles } from "stores/backend"
+  import RoleSelect from "components/common/RoleSelect.svelte"
+
+  export let group
+
+  $: appOptions = $apps.map(app => ({
+    label: app.name,
+    value: app,
+  }))
+  $: prodAppId = selectedApp ? apps.getProdAppID(selectedApp.appId) : ""
+  $: confirmDisabled =
+    (!selectingRole && !selectedApp) || (selectingRole && !selectedRoleId)
+  let selectedApp, selectedRoleId
+  let selectingRole = false
+
+  async function appSelected() {
+    if (!selectingRole) {
+      selectingRole = true
+      await roles.fetchByAppId(prodAppId)
+      // return false to stop closing modal
+      return false
+    } else {
+      await groups.actions.addApp(group._id, prodAppId, selectedRoleId)
+    }
+  }
+</script>
+
+<ModalContent
+  onConfirm={appSelected}
+  size="M"
+  title="Add app to group"
+  confirmText="Next"
+  showSecondaryButton
+  secondaryButtonText="Back"
+  secondaryAction={() => (selectingRole = false)}
+  disabled={confirmDisabled}
+>
+  {#if !selectingRole}
+    <Body>Select the app to add to the <i>"{group.name}"</i> group.</Body>
```

Comment:
Since users are the things in the groups, I would update this copy here to be more explicit.

`Select an app the assign permissions for <Group name>`

Reply:
![image](https://user-images.githubusercontent.com/4407001/192340866-d760c38b-8ebc-42f2-b7e8-e89ea0140c59.png)